tests/Doctrine/Tests/Mocks/HydratorMockResult.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\Mocks;
+
+use BadMethodCallException;
+use Doctrine\DBAL\Result;
+use Traversable;
+
+use function count;
+use function current;
+use function next;
+use function reset;
+
+class HydratorMockResult implements Result
+{
+    /** @var array<int, array<string, mixed>> */
```

Comment:
If the `int` does not matter, I'd use `list` here: although it's not required this be a list, cases where people would not pass a list are rare enough and IMO it's better if people reading that piece of code can understand that the key does not matter.

Reply:
`list<>`  is more accurate. I've changed the annotation.